Source/ACE/Entity/Generator.cs

```diff
@@ -0,0 +1,100 @@
+﻿using ACE.Entity.Enum;
+using ACE.Network.Enum;
+using ACE.Network.Motion;
+using System;
+using System.Diagnostics;
+
+namespace ACE.Entity
+{
+    public class Generator : WorldObject
+    {
+        public Generator(ObjectGuid guid, AceObject baseAceObject)
+            : base((ObjectType)baseAceObject.ItemType, guid)
+        {
+            Name = baseAceObject.Name ?? "NULL";
```

Comment:
Started from DebugObject and then hard coded propertied I'm assuming to be immutable between generators and likely could be left out of the database in the future entirely. Left it commented for easier undo, can remove in the future when certain of the path. I'm also making the assumption that code from GeneratorFactory eventually ends up in this class instead and we make use of actionchaining to spawn/despawn, etc